src/Core/src/Credentials/ChainProvider.php

```diff
@@ -78,6 +78,7 @@ public static function createDefaultChain(?HttpClientInterface $httpClient = nul
             new WebIdentityProvider($logger, null, $httpClient),
             new IniFileProvider($logger, null, $httpClient),
             new ContainerProvider($httpClient, $logger),
+            new PodIdentityProvider($httpClient, $logger),
             new InstanceProvider($httpClient, $logger),
```

Comment:
not related to your PR, but I discovered that AWS uses either `EscProvider` or `InstanceProvider` in the default chain:

```
        if (self::shouldUseEcs()) {
            $defaultChain['ecs'] = self::ecsCredentials($config);
        } else {
            $defaultChain['instance'] = self::instanceProfile($config);
        } 
```

I think we could safely do the same :thinking: